src/uct/ib/ud/base/ud_ep.c

```diff
@@ -257,8 +259,10 @@ static unsigned uct_ud_ep_deferred_timeout_handler(void *arg)
         return 0;
     }
 
-    if (ep->flags & UCT_UD_EP_FLAG_PRIVATE) {
+    if (!(ep->flags & UCT_UD_EP_FLAG_TX)) {
         ucs_assert(ucs_queue_is_empty(&ep->tx.window));
+        uct_ud_iface_cep_remove_ep(iface, ep);
+        ep->flags &= ~UCT_UD_EP_FLAG_RX;
```

Comment:
don't we need to do it even if ep->flags & UCT_UD_EP_FLAG_TX is 0?
so if we have timeout error no need to send FIN